Branch pokedex

Ulises-Berasain opened the pull request.

Ulises-Berasain committed:
Pokedex

pokedex api funcional

Ulises-Berasain committed:
Update package.json

Ulises-Berasain committed:
jquery

fsodano commented on index.html:
```diff
@@ -0,0 +1,81 @@
+    <meta http-equiv='X-UA-Compatible' content='IE=edge'>
+    <title>Pokedex</title>
+    <meta name='viewport' content='width=device-width, initial-scale=1'>
+    <link rel='stylesheet' type='text/css' media='screen' href='estilos.css'>
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" integrity="sha384-JcKb8q3iqJ61gNV9KGb8thSsNjpSL0n8PARn9HuZOnIxN0hoP+VmmDGMN5t9UJ0Z" crossorigin="anonymous">
```
primero van los 3rd party y luego tu código, es decir: Primero bootstrap y luego tu código.

fsodano commented on package.json:
```diff
@@ -0,0 +1,30 @@
+  "version": "1.0.0",
+  "description": "Pokedex",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "cypress-dev": "cypress open"
+  },
```
tus scripts deberían incluir los comandos para levantar tu servidor `http-server` ponele

fsodano commented on package.json:
```diff
@@ -0,0 +1,30 @@
+    "popper.js": "^1.16.1"
+  },
+  "dependencies": {
+    "eslint": "^7.7.0",
```
esta es un devDependency

fsodano commented on package.json:
```diff
@@ -0,0 +1,30 @@
+  "homepage": "https://github.com/Ulises-Berasain/pokedex-api#readme",
+  "devDependencies": {
+    "cypress": "^4.12.1",
+    "popper.js": "^1.16.1"
```
esta es una dependency de bootstrap, no?

fsodano commented on package.json:
```diff
@@ -0,0 +1,30 @@
+  "dependencies": {
+    "eslint": "^7.7.0",
+    "jquery": "^1.12.4",
+    "@types/jquery": "^3.5.1",
```
esta es una devDependency

fsodano commented on src/botones.js:
```diff
@@ -0,0 +1,62 @@
+                $("#error").html("");
+            })
+            .catch(error => {
+                $("#error").html("").html("Error! ingresar pokemon o ID validos y que su nombre este minuscula");
```
piola 💯

Ulises-Berasain committed:
Tabla

Agrego una tabla para poder mostrar los ataques y version de cada uno para tener mas datos de los pokes

Ulises-Berasain committed:
Correccion de tabla

Arreglo el .innerHTML faltante y modifico un poco el codigo.

Ulises-Berasain committed:
Tabla ya funcionando

Ahora muestra bien las celdas de los ataques al elegir un pokemon.

Ulises-Berasain committed:
Modulos

Organizo mejor los modulos y agrego un proximo local storage

Ulises-Berasain committed:
Modulos

Arreglo el boton de busqueda y ademas organizo un poco mejor los modulos.

Ulises-Berasain committed:
Correcciones

cambio los "/src" por "../" y corrigo un "{" que no iba del fetch en cargarPokemons

Ulises-Berasain committed:
Responsive

Agrego css para que pueda ser una pagina visible en celulares, tabletas o desde una PC misma.

Ulises-Berasain committed:
LocalStorage y fondo imagen pokemon

Agrego LocalStorage, arreglo el fondo de los pokemon en la tarjeta ademas de evitar otros errores si se busca el Pokemon con mayuscula para causarle menos problemas al usuario.

Ulises-Berasain committed:
Cypress

Testeo de la pagina usando Cypress
